subsys/task_wdt/task_wdt_shell.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024 Vogl Electronic GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <zephyr/device.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/task_wdt/task_wdt.h>
+
+static int cmd_init(const struct shell *sh, size_t argc, char *argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+#ifdef CONFIG_TASK_WDT_HW_FALLBACK
+	const struct device *const wdt = DEVICE_DT_GET(DT_ALIAS(watchdog0));
+#else
+	const struct device *const wdt = NULL
+#endif
+
+	shell_fprintf(sh, SHELL_INFO, "Init task watchdog ...\n");
+
+	int ret = task_wdt_init(wdt);
+
+	if (ret < 0) {
+		shell_fprintf(sh, SHELL_ERROR, "Failed to init task watchdog: %d\n", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int cmd_add(const struct shell *sh, size_t argc, char **argv)
+{
+	if (argc < 2) {
```

Comment:
Does this not need *exactly* one parameter?

```suggestion
	if (argc != 2) {
```